.github/workflows/shell.yml

```diff
@@ -0,0 +1,36 @@
+name: Shell
+
+# Relevant to events - https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows
+on: 
+  pull_request:
+    types: [synchronize, opened, reopened, ready_for_review]
+    paths:
+    - '**.sh'
+
+jobs:
+  # Linting
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Installing dependencies..
+        run: if ! apt list --installed | grep -qP ".*shellcheck.*"; then apt install -y shellcheck; fi
+      - name: Linting..
+        run: |
+          cd "$GITHUB_WORKSPACE" # Change dir in required workspace
+
+          # Process files
+          for file in $(git ls-tree --name-only -r ${{ github.sha }} | grep '\.bash$'); do
```

Comment:
This might be commonly recommended, but it's still contrary to the practices recommended by shellcheck.

Instead of « for var in $(…) », use « while read -r var … »